controllers/handlers.go

```diff
@@ -110,11 +111,16 @@ func HandleUpload(uploader TokenUploader) func(http.ResponseWriter, *http.Reques
 		}
 
 		if err := uploader.Upload(ctx, tokenObjectName, tokenObjectNamespace, data); err != nil {
-			if kubeErrors.IsNotFound(err) { // supports wrapped errors
+			switch errors.ReasonForError(err) {
+			case v1.StatusReasonNotFound:
 				LogErrorAndWriteResponse(r.Context(), w, http.StatusNotFound, "specified SPIAccessToken does not exist", err)
-				return
+			case v1.StatusReasonForbidden:
+				LogErrorAndWriteResponse(r.Context(), w, http.StatusForbidden, "authorization token does not have permission to update SPIAccessToken", err)
+			case v1.StatusReasonUnauthorized:
+				LogErrorAndWriteResponse(r.Context(), w, http.StatusUnauthorized, "invalid authorization token, cannot update SPIAccessToken", err)
+			default:
+				LogErrorAndWriteResponse(r.Context(), w, http.StatusInternalServerError, "failed to upload the token", err)
```

Comment:
I was thinking about changing the default to something else than very generic 500 and return 500 only if `StatusReasonInternalError`, but I couldn't find better response code. Maybe `Bad Request` ?